docs/walk-through/loops.md

```diff
@@ -2,13 +2,41 @@
 
 When writing workflows, it is often very useful to be able to iterate over a set of inputs, as this is how argo-workflows can perform loops.
 
-There are two basic ways of running a template multiple times.
+There are three basic ways of running a template multiple times.
 
+- `withSequence` iterates over a sequence of numbers.
 - `withItems` takes a list of things to work on. Either
     - plain, single values, which are then usable in your template as '{{item}}'
     - a JSON object where each element in the object can be addressed by it's key as '{{item.key}}'
 - `withParam` takes a JSON array of items, and iterates over it - again the items can be objects like with `withItems`. This is very powerful, as you can generate the JSON in another step in your workflow, so creating a dynamic workflow.
 
+## `withSequence` example
+
+This runs a template multiple times using `withSequence`.
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: Workflow
+metadata:
+  generateName: loop-sequence-
+spec:
+  entrypoint: loop-sequence-example
+
+  templates:
+  - name: loop-sequence-example
+    steps:
+    - - name: hello-world-x5
+        template: hello-world
+        withSequence:
+          count: "5"
+
+  - name: run-pod
+    container:
+      image: argoproj/argosay:v2
+      command: ["sh", -c]
+      args: ["echo hello world!"]
```

Comment:
```suggestion
      image: docker/whalesay:latest
      command: [cowsay]
      args: ["hello world!"]
```

I mentioned using `whalesay` in my [previous comment](https://github.com/argoproj/argo-workflows/pull/12577#discussion_r1495373464) for consistency with the `withItems` example.

Also `argosay` prints things (it uses [`cowsay`](https://github.com/argoproj/argo-workflows/blob/a68ea0feabc87c09d5e13d12e6f0d1a61adc5b16/test/e2e/images/argosay/v1/Dockerfile#L17), so it doesn't need a `sh -c echo`.